pkg/action/action.go

```diff
@@ -418,3 +418,8 @@ func (c *Configuration) Init(getter genericclioptions.RESTClientGetter, namespac
 
 	return nil
 }
+
+func warning(format string, v ...interface{}) {
+	format = fmt.Sprintf("WARNING: %s\n", format)
+	fmt.Fprintf(os.Stderr, format, v...)
```

Comment:
If I refer to the action package and want to change the error output stream, what do I need to do?

Reply:
> I simply copied that from cmd/helm/helm.go
> 
> That's used in:
> 
> cmd/helm/helm.go
> 
> ```
> func main() {
>         actionConfig := new(action.Configuration)
>         cmd, err := newRootCmd(actionConfig, os.Stdout, os.Args[1:])
>         if err != nil {
>                 warning("%+v", err)
>                 os.Exit(1)
>         }
> ```
> 
> And several other places in package `main` and the implementation is exactly the same:
> 
> ```
>  56 func warning(format string, v ...interface{}) {
>  57         format = fmt.Sprintf("WARNING: %s\n", format)
>  58         fmt.Fprintf(os.Stderr, format, v...)
>  59 }
> ```
> 
> I'd really prefer to get and pass func Log from somewhere, but I don't know how to pass a Log method into `func (c *ChartPathOptions) LocateChart(name string, settings *cli.EnvSettings) (string, error) {` without making this a really complex change.

The action package will be called by people, it is not the same as the cmd package. So I don't think it can be copied directly over